src/usage_metrics/raw/s3.py

```diff
@@ -12,31 +14,40 @@
 from tqdm import tqdm
 
 BUCKET_URI = "pudl-s3-logs.catalyst.coop"
-LOCAL_DIR = "data/pudl_s3_logs/"
+PATH_EXT = "data/pudl_s3_logs/"
 # if not Path.exists(Path(LOCAL_DIR)):
 #     Path.mkdir(LOCAL_DIR)
 
 
 def download_s3_logs_from_gcs(
+    context: AssetExecutionContext,
     partition_dates: tuple[str],
 ) -> list[Path]:
     """Download all logs from GCS bucket.
 
     If the file already exists locally don't download it.
     """
+    # Determine where to save these files
+    download_dir = Path(
+        os.environ.get("DATA_DIR", TemporaryDirectory().name), "pudl_s3_logs/"
+    )
+    if not Path.exists(download_dir):
+        Path.mkdir(download_dir, parents=True)
```

Comment:
I think this subtly and surprisingly makes a non-temporary directory with a temporary-looking name.

Once the `TemporaryDirectory` object is deleted, the directory itself is also deleted - and since you never assign the `TemporaryDirectory` to a variable name, the Python runtime deletes the `TemporaryDirectory` right after it's been created. Check this out:

```python
>>> from tempfile import TemporaryDirectory
>>> p = TemporaryDirectory().name                                                                                                              
>>> p
'/var/folders/7q/28j531_d151fn0d3831ql0c00000gr/T/tmpuzvg0kuq'
>>> from pathlib import Path
>>> Path(p)
PosixPath('/var/folders/7q/28j531_d151fn0d3831ql0c00000gr/T/tmpuzvg0kuq')
>>> Path(p).exists()
False
```

I think that whole "make this directory if it doesn't exist" block you have immediately afterwards obscures that deletion.

The typical way to manage this is with a context manager at the call site:

```python
def raw_s3_logs(...):
    ...
    with TemporaryDirectory() as td:
        file_paths = download_s3_logs_from_gcs(context=..., partition_dates=..., download_dir=....)
    ...
```
Which then deletes the directory after exiting the context manager.

But if you don't want to change the function signature here, you could also work around this by assigning that `TemporaryDirectory` to a local variable inside this function body:

```python
def download_s3_logs_from_gcs(...):
    ...
    td = TemporaryDirectory()
    download_dir = ...
    ...
    td.cleanup()
```
Technically, since `td` will get deleted when you leave the function, you don't *have* to manually `cleanup()`. But it makes the intent explicit which is nice. And it spares people the effort of wondering if this temporary dir got cleaned up.

Reply:
Ooh, good catch. I've resolved this.